src/Event.php

```diff
@@ -34,11 +49,13 @@ class Event implements EventInterface
      *
      * Accept a target and its parameters.
      *
-     * @param  string $name Event name
-     * @param  string|object $target
-     * @param  array|ArrayAccess $params
+     * @param string|null $name Event name
+     * @param string|object|null $target
+     * @psalm-param TTarget $target
+     * @param array|ArrayAccess|object|null $params
+     * @psalm-param TParams|null $params
```

Comment:
I think that would be better yeah, paired with the reversal of the ctor `null` change.

The suppression for `protected $params` is because `array<empty, empty>` cannot be assigned to `TParams` . There seems to be no way to infer an empty array for `TParams` when the ctor  receives `null` as the third argument.

I could fix that with `$this->setParams($params ?? [])`, but that would change behavior when someone has extended `Event` without calling the parent's ctor. So I figured the suppression was a better solution.